ui/selectors/selectors.js

```diff
@@ -1086,9 +1087,32 @@ export function getIsTokenNetworkFilterEqualCurrentNetwork(state) {
   return false;
 }
 
+/**
+ * Returns an object indicating which networks
+ * tokens should be shown on in the portfolio view.
+ *
+ * @param {*} state
+ * @returns {Record<Hex, boolean>}
+ */
 export function getTokenNetworkFilter(state) {
+  const currentChainId = getCurrentChainId(state);
   const { tokenNetworkFilter } = getPreferences(state);
-  return tokenNetworkFilter || {};
+
+  // Portfolio view not enabled outside popular networks
+  if (!FEATURED_NETWORK_CHAIN_IDS.includes(currentChainId)) {
+    return { [currentChainId]: true };
+  }
+
+  // Portfolio view only enabled on featured networks
+  return Object.entries(tokenNetworkFilter || {}).reduce(
+    (acc, [chainId, value]) => {
+      if (FEATURED_NETWORK_CHAIN_IDS.includes(chainId)) {
+        acc[chainId] = value;
+      }
+      return acc;
+    },
+    {},
+  );
```

Comment:
Should we care about the PORTFOLIO_VIEW feature flag here? Seems like we probably don't need to

Reply:
i can add that https://github.com/MetaMask/metamask-extension/pull/29071/commits/2010b0202a5fe153a75ed38b1610171d9722a7a0